packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -117,11 +118,15 @@ class AbstractModal extends React.Component {
       if (!mainDocumentElement.hasAttribute('data-overlay-count')) {
         mainDocumentElement.setAttribute('data-overlay-count', '1');
         mainDocumentElement.setAttribute('inert', '');
-        // Shift focus to modal when opened
-        this.modalElement.current.focus();
       } else if (mainDocumentElement && mainDocumentElement.hasAttribute('data-overlay-count')) {
         mainDocumentElement.setAttribute('data-overlay-count', `${inert + 1}`);
-        this.modalElement.current.focus();
+      }
+      // Shift focus to first tabbable element in modal
+      if (tabbable(this.modalElement.current)[0]) {
+        tabbable(this.modalElement.current)[0].focus();
+      } else {
+        // If no focusable elements are withing the modal, shift to modal dialog begin help text
+        this.modalElement.current.querySelector('[data-terra-abstract-modal-begin]').focus();
       }
```

Comment:
I ran into this when working on the select's mobile modal as well, but I like this logic. Seems about as safe as we can get.

However, I think we'll still run into some issues on iOS VoiceOver, but they're gonna be really hard to combat. As in, the only thing we can really do is log a bug to Apple.

I believe you've seen this as well, but iOS VoiceOver doesn't always listen to you when you say

>"focus on this element, it has a tabIndex set".

It seems to only like to move the virtual indicator to interactable elements, like buttons, anchors, etc, when programmatically manipulating the focus..

Reply:
Looking below, I see that `data-terra-abstract-modal-begin` has a `tabIndex="-1"`. So, what happens here with VoiceOver when there's not a `tabbable` element in the modal (should be rare, hopefully)?

Reply:
You're correct, VoiceOver on iOS is very picky about which elements it shifts the virtual indicator to when you programmatically shifting focus. I've found from [testing this page](https://codepen.io/backwardok/pen/xpPZKL) that VoiceOver on iOS will shift focus to a `span` or `div` with `tabIndex="-1"` set on it when you programmatically shift focus to the element **if** the `span` / `div` only contains a string of text within it. 

If you nest other elements within the `span` or `div`, then VoiceOver on iOS doesn't seem to allow shifting the virtual indicator to a `span` or `div` with `tabIndex="-1"` on via programmatic focus shift. I agree that this seems like an Apple bug.

Based on this, I do have logic that will account for this without making the code to purpose built for VoiceOver. When the modal opens, we look for interactable elements via tabbable, and focus the first one we find within the modal, if in the rare case people don't include any interactable elements, focus will shift to visually hidden text with the tabIndex="-1" on it that indicates the beginning of the modal dialog. This allows the focus to move into the modal on iOS when using VoiceOver even if people forget or choose not to add interactable elements to the modal.

This can be tested on the following page: https://terra-framework-deploye-pr-771.herokuapp.com/#/tests/terra-abstract-modal/abstract-modal/abstract-modal-no-focusable-content

Reply:
Interesting.. I guess that would explain why when using iOS VoiceOver with the `terra-select`, it still navigates to the visually hidden text

![image](https://user-images.githubusercontent.com/34115417/61149187-2c85bf80-a4a6-11e9-8c38-38a93bab4afa.png)

Here it's focused (for lack of a better word) on this [hidden "Selected" text](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/_Frame.jsx#L229-L231)

Reply:
Yeah, I believe the virtual indicator will jump to DOM elements containing text as the user swipes left and right on iOS when using VoiceOver. Since that visually hidden text is in the DOM, VoiceOver will move the virtual indicator to it as the user navigates the page with left/right swipe gestures.